packages/approval-controller/src/ApprovalController.ts

```diff
@@ -612,6 +652,50 @@ export class ApprovalController extends BaseControllerV2<
     });
   }
 
+  /**
+   * Starts a new approval flow.
+   *
+   * @param opts - Options bag.
+   * @param opts.id - The id of the approval flow.
+   * @returns The object containing the approval flow id.
+   */
+  startFlow(opts: ApprovalFlowOptions = {}): ApprovalFlowStartResult {
+    const id = opts.id ?? nanoid();
+    const finalOptions = { id };
+
+    this.update((draftState) => {
+      draftState.approvalFlows.push(finalOptions);
+    });
+
+    this._showApprovalRequest();
+
+    return { id };
+  }
+
+  /**
+   * Ends the current approval flow.
+   *
+   * @param flowId - The id of the approval flow to end.
+   */
+  endFlow(flowId: string) {
```

Comment:
I was torn whether this warranted an `EndFlowOptions` to simply further extensions otherwise we'd have to support multiple overloads. Perhaps better to start off with the object?

Reply:
It will save us from introducing breaking changes if we add any feature, so yes, let's go with an options bag.